### PR TITLE
Find MySQL 5.5 installation on CentOS

### PR DESCRIPTION
**Pull request by jjagodzinski**
MySQL from CentOS SCL repository is installed in different location. You should also check for binaries in this location.
Without this change puppet returns error like: Could not find a suitable provider for mysql_datadir

### CODE REVIEW

File: lib/puppet/provider/mysql_datadir/mysql.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@
   initvars
 
   # Make sure we find mysqld on CentOS
-  ENV['PATH']=ENV['PATH'] + ':/usr/libexec'
+  ENV['PATH']=ENV['PATH'] + ':/opt/rh/mysql55/root/usr/bin:/opt/rh/mysql55/root/usr/libexec:/usr/libexec'
```
Comment (igalic) [inline lib/puppet/provider/mysql_datadir/mysql.rb]: how does this work on *BSD?

Comment (jjagodzinski) [inline lib/puppet/provider/mysql_datadir/mysql.rb]: I didn't test it on *BSD. I don't have *BSD in my environment.

**Comment by jonnytpuppet** [comment]
I like the fact that this issue has a "simple" solution, although it is not elegant to pollute an environment's PATH with paths that clearly won't exist.

Keen to hear from @hunner if you can think of a nearly way to do this?

Thanks!

File: lib/puppet/provider/mysql_datadir/mysql.rb (hunk 1 above)
Comment (daenney) [inline lib/puppet/provider/mysql_datadir/mysql.rb]: How does this commit warrant asking about BSD? It adds a path extremely specific to Red Hat Collections, `/opt/rh`.

**Comment by hunner** [comment]
Could you rebase because we merged #838? And also, would it work to put the custom paths AFTER /usr/libexec instead of before? To limit side-effects of this change on existing installs.

**Comment by jjagodzinski** [comment]
New change in #842 (I've got some problems with rebase).

**Pull request closed without merging**